Core.UnitTests/MutableReflection/Implementation/MemberFactory/MethodOverrideFactoryTest.cs

```diff
@@ -407,35 +466,43 @@ public void GetOrAddImplementation_InterfaceMethod_InvalidCandidate ()
       methodFactoryMock
-          .Expect (
+          .Setup (
               mock => mock.CreateMethod (
-                  Arg.Is (mutableType),
-                  Arg.Is (expectedOverrideMethodName),
-                  Arg.Is (expectedOverrideAttributes | MethodAttributes.Virtual | MethodAttributes.HideBySig),
-                  Arg.Is (GenericParameterDeclaration.None),
-                  Arg<Func<GenericParameterContext, Type>>.Is.Anything,
-                  Arg<Func<GenericParameterContext, IEnumerable<ParameterDeclaration>>>.Is.Anything,
-                  Arg<Func<MethodBodyCreationContext, Expression>>.Is.Anything))
-          .WhenCalled (
-              mi =>
+                  mutableType,
+                  expectedOverrideMethodName,
+                  (expectedOverrideAttributes | MethodAttributes.Virtual | MethodAttributes.HideBySig),
+                  It.Is<IEnumerable<GenericParameterDeclaration>> (param => param.SequenceEqual (GenericParameterDeclaration.None)),
+                  It.IsAny<Func<GenericParameterContext, Type>>(),
+                  It.IsAny<Func<GenericParameterContext, IEnumerable<ParameterDeclaration>>>(),
+                  It.IsAny<Func<MethodBodyCreationContext, Expression>>()))
+          .Callback (
+              (
+                  MutableType declaringType,
+                  string name,
+                  MethodAttributes attributes,
+                  IEnumerable<GenericParameterDeclaration> genericParameters,
+                  Func<GenericParameterContext, Type> returnTypeProvider,
+                  Func<GenericParameterContext, IEnumerable<ParameterDeclaration>> parameterProvider,
+                  Func<MethodBodyCreationContext, Expression> bodyProvider) =>
               {
-                var returnType = ((Func<GenericParameterContext, Type>) mi.Arguments[4]) (_noGenericParameters);
-                var parameters = ((Func<GenericParameterContext, IEnumerable<ParameterDeclaration>>) mi.Arguments[5]) (_noGenericParameters);
+                var returnType = ((Func<GenericParameterContext, Type>) returnTypeProvider) (_noGenericParameters);
+                var parameters = ((Func<GenericParameterContext, IEnumerable<ParameterDeclaration>>) parameterProvider) (_noGenericParameters);
                 Assert.That (returnType, Is.SameAs (typeof (void)));
                 var parameter = parameters.Single();
                 Assert.That (parameter.Name, Is.EqualTo (expectedParameterName));
                 Assert.That (parameter.Type, Is.SameAs (typeof (int)));
 
                 if (skipBodyProviderCheck)
                   return;
-                var body = ((Func<MethodBodyCreationContext, Expression>) mi.Arguments[6]) (CreateMethodBodyCreationContext (baseMethod));
+                var body = ((Func<MethodBodyCreationContext, Expression>) bodyProvider) (CreateMethodBodyCreationContext (baseMethod));
```

Comment:
Those casts are extranous. please check the other MemberFactory test-classes too.